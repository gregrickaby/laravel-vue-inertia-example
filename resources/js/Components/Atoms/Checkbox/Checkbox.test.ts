import Checkbox from '@/Components/Atoms/Checkbox/Checkbox.vue'
import { mount } from '@vue/test-utils'

/**
 * Define a test suite.
 */
describe('Checkbox', () => {
    /**
     * Mount the component.
     */
    const wrapper = mount(Checkbox, {
        props: {
            checked: false
        }
    })

    /**
     * Assert the component input renders correctly.
     */
    test('renders a checkbox input', () => {
        const checkbox = wrapper.find('input[type="checkbox"]')
        expect(checkbox.exists()).toBe(true)
    })

    /**
     * Assert component toggles its state and emits an event on state change.
     */
    test('toggles checked state and emits update:checked event', async () => {
        const checkbox = wrapper.find('input[type="checkbox"]')
        if (checkbox.element instanceof HTMLInputElement) {
            // Checkbox: checked
            await checkbox.setValue(true)
            expect(checkbox.element.checked).toBe(true)
            expect(wrapper.emitted('update:checked')).toBeTruthy()
            expect(wrapper.emitted('update:checked')?.[0]).toEqual([true])

            // Checkbox: unchecked
            await checkbox.setValue(false)
            expect(checkbox.element.checked).toBe(false)
            expect(wrapper.emitted('update:checked')?.[1]).toEqual([false])
        }
    })

    /**
     * Assert the component matches the snapshot.
     */
    test('checkbox matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
