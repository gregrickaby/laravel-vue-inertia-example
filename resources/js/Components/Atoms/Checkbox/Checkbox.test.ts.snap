// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox > checkbox matches snapshot 1`] = `"<input type="checkbox" class="rounded border-gray-300 text-indigo-600 shadow-sm focus:ring-indigo-500 dark:border-gray-700 dark:bg-gray-900 dark:focus:ring-indigo-600 dark:focus:ring-offset-gray-800">"`;
